src/System.IO.Compression/src/System/IO/Compression/ZipCustomStreams.cs

```diff
@@ -20,11 +20,11 @@ internal class WrappedStream : Stream
 
         #region constructors
 
-        internal WrappedStream(Stream baseStream, Boolean canRead, Boolean canWrite, Boolean canSeek, EventHandler onClosed)
-            : this(baseStream, canRead, canWrite, canSeek, false, onClosed)
+        internal WrappedStream(Stream baseStream, Boolean closeBaseStream)
+            : this(baseStream, true, true, true, closeBaseStream, null)
         { }
 
-        internal WrappedStream(Stream baseStream, Boolean canRead, Boolean canWrite, Boolean canSeek, Boolean closeBaseStream, EventHandler onClosed)
+        private WrappedStream(Stream baseStream, Boolean canRead, Boolean canWrite, Boolean canSeek, Boolean closeBaseStream, EventHandler onClosed)
```

Comment:
Is WrappedStream only ever constructed with ```true``` for ```canRead``, ```canWrite```, and ```canSeek```?  If so, can we just get rid of those parameters and associated fields on the stream instance?  If the only reason they still exist is to be able to start returning false from ```Can*``` after disposal, ```_isDisposed``` could be checked in the ```Can*``` properties instead of those fields, right?